### ISSUE TITLE
Link widget requires roles, ARIA properties, and keyboard support (Accessibility)

### ISSUE THREAD

**Issue by jcollins-disney** [flutter/flutter#157210]
### Use case

The Link widget requires roles, ARIA properties, and expected keyboard support to meet accessibility compliance requirements outlined in the Web Content Accessibility Guidelines (WCAG).

### WAI-ARIA Roles, States, and Properties
- The element containing the link text or graphic is marked up as `<a>` or has `role="link"`
- If the link requires an additional programmatic label to give context to assistive technology users, use `aria-label` or `aria-labelledby="[IDREF]"`, with [IDREF] referencing the unique ID of the element that describes the purpose of the tab group

**Comment by iapicca** [comment]
the topic seems to be addressed this [document](https://docs.google.com/document/d/1VRagJfHEbehVPa023xGUcFTqe5LgYGU2wYBmnNkwVmA/edit)
as mentioned in this issue
- https://github.com/flutter/flutter/issues/107861#issuecomment-2066989390

**Comment by darshankawar** [comment]
The linked document mentions Link widget. @jcollins-disney I will close this in favor of the linked issue. Please follow-up in it for further updates. If you disagree, write in comments and I'll reopen it.

**Comment by darshankawar** [comment]
Reopening based on https://github.com/flutter/flutter/issues/107861#issuecomment-2427190284

**Comment by chunhtai** [comment]
I thought this is already supported? cc @yjbanov

**Comment by yjbanov** [comment]
The ARIA attributes go into the semantics tree, not the widget itself. However, let's double-check, in case we're missing something.

**Comment by mdebbar** [comment]
The Link widget is currently implemented as an `<a>` element, which satisfies the request in this issue. Plus there were several semantics improvements for the Link widget here: https://github.com/flutter/packages/pull/6711

I'm closing this issue but please feel free to reopen if there's still something missing.
